src/main/java/net/glowstone/constants/GlowParticle.java

```diff
@@ -13,53 +16,55 @@
 public final class GlowParticle {
 
     private static final int[] EMPTY = new int[0];
-    private static final int[] ids = new int[Effect.values().length];
+    private static final int[] ONE_EMPTY = new int[]{0};
+    private static final int[] TWO_EMPTY = new int[]{0, 0};
+
+    private static final Map<Object, Integer> ids = new HashMap<>();
```

Comment:
Maintainability, readability and also to support the Particle enum without having to create another array.